File: src/main/java/org/stt/gui/jfx/binding/STTBindings.java
package org.stt.gui.jfx.binding;

import javafx.beans.binding.Bindings;
import javafx.beans.value.ObservableStringValue;
import javafx.beans.value.ObservableValue;
import org.stt.time.DateTimes;

import java.time.Duration;

public class STTBindings {
    private STTBindings() {
    }

    public static ObservableStringValue formattedDuration(final ObservableValue<Duration> duration) {
        return Bindings.createStringBinding(() -> DateTimes.FORMATTER_PERIOD_HHh_MMm_SSs.print(duration.getValue()), duration);
    }
}

File: src/main/kotlin/org/stt/gui/jfx/binding/STTBindings.kt
package org.stt.gui.jfx.binding

import javafx.beans.binding.Bindings
import javafx.beans.value.ObservableStringValue
import javafx.beans.value.ObservableValue
import org.stt.time.DateTimes

import java.time.Duration
import java.util.concurrent.Callable

object STTBindings {

    fun formattedDuration(duration: ObservableValue<Duration>): ObservableStringValue {
        return Bindings.createStringBinding(Callable { DateTimes.FORMATTER_PERIOD_HHh_MMm_SSs(duration.value) }, duration)
    }
}
